client/src/plus/ai-help/use-ai.ts

```diff
@@ -407,21 +408,23 @@ export function useAiChat({
     });
   }, [setSearchParams, chatId]);
 
-  const clearParamsFromUrl = () => {
-    window.history.replaceState(
-      "",
-      "",
-      window.location.protocol +
-        "//" +
-        window.location.host +
-        window.location.pathname
+  const removeChatIdFromUrl = () => {
+    const searchParams = new URLSearchParams(window.location.search);
+
+    searchParams.delete("c");
+
+    navigate(
+      {
+        search: searchParams.toString(),
+      },
+      { replace: true }
     );
```

Comment:
I think it might be possible to just use `useSearchParams` if the docs can be believed:
> The setSearchParams function works like [navigate](https://reactrouter.com/en/main/hooks/use-navigate), but only for the [search portion](https://developer.mozilla.org/en-US/docs/Web/API/Location/search) of the URL. Also note that the second arg to setSearchParams is the same type as the second arg to navigate.
https://reactrouter.com/en/main/hooks/use-search-params

Something like:
```js
const [_, setSearchParams] = useSearchParams();

...

setSearchParams(
  (searchParams) => {
    const x = new URLSearchParams(searchParams)
    x.delete("c")
    return x
  },
  { replace: true }
)
```